dotnet/src/VectorDataIntegrationTests/VectorDataIntegrationTests/DependencyInjectionTests.cs

```diff
@@ -56,64 +69,104 @@ public void CollectionNameCantBeNullOrEmpty()
+            services.Add(new ServiceDescriptor(typeof(VectorStore), serviceKey, (sp, key) => new FakeVectorStore(), lifetime));
 
-        this.RegisterVectorStore(services, lifetime, serviceKey);
+            registrationDelegate(services, serviceKey, lifetime);
 
-        using ServiceProvider serviceProvider = services.BuildServiceProvider();
-        // let's ensure that concrete types are registered
-        Verify<TVectorStore>(serviceProvider, lifetime, serviceKey);
+            using ServiceProvider serviceProvider = services.BuildServiceProvider();
+            // let's ensure that concrete types are registered
+            Verify<TVectorStore>(serviceProvider, lifetime, serviceKey);
+        }
     }
 
     [Theory]
     [MemberData(nameof(LifetimesAndServiceKeys))]
     public void CanRegisterConcreteTypeCollectionsAfterSomeAbstractionHasBeenRegistered(ServiceLifetime lifetime, object? serviceKey)
     {
-        IServiceCollection services = this.CreateServices(serviceKey);
+        foreach (var registrationDelegate in this.CollectionDelegates)
+        {
+            IServiceCollection services = this.CreateServices(serviceKey);
+
+            // Users may be willing to register more than one VectorStoreCollection implementation.
+            services.Add(new ServiceDescriptor(typeof(VectorStoreCollection<TKey, TRecord>), serviceKey, (sp, key) => new FakeVectorStoreRecordCollection(), lifetime));
+
+            registrationDelegate(services, serviceKey, "collectionName", lifetime);
+
+            using ServiceProvider serviceProvider = services.BuildServiceProvider();
+            // let's ensure that concrete types are registered
+            Verify<TCollection>(serviceProvider, lifetime, serviceKey);
+        }
+    }
+
+    [Theory]
+    [MemberData(nameof(LifetimesAndServiceKeys))]
+    public void EmbeddingGeneratorIsResolved(ServiceLifetime lifetime, object? serviceKey)
+    {
+        foreach (var registrationDelegate in this.CollectionDelegates)
+        {
+            IServiceCollection services = this.CreateServices(serviceKey);
 
-        // Users may be willing to register more than one VectorStoreCollection implementation.
-        services.Add(new ServiceDescriptor(typeof(VectorStoreCollection<TKey, TRecord>), serviceKey, (sp, key) => new FakeVectorStoreRecordCollection(), lifetime));
+            bool wasResolved = false;
+            services.AddSingleton<IEmbeddingGenerator>(sp =>
+            {
+                wasResolved = true;
+                return null!;
+            });
 
-        this.RegisterCollection(services, lifetime, serviceKey: serviceKey);
+            registrationDelegate(services, serviceKey, "collectionName", lifetime);
 
-        using ServiceProvider serviceProvider = services.BuildServiceProvider();
-        // let's ensure that concrete types are registered
-        Verify<TCollection>(serviceProvider, lifetime, serviceKey);
+            Assert.False(wasResolved); // it's lazy
```

Comment:
This seems to be testing general DI flow that isn't related to MEVD

Reply:
My purpose was to show that we don't create the object when the extension method gets called, but rather register a factory method that is lazy executed